app/controllers/api/v2/qc_files_controller.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Api
+  module V2
+    # Provides a JSON API controller for QcFile
+    # See: http://jsonapi-resources.com/ for JSONAPI::Resource documentation
+    class QcFilesController < JSONAPI::ResourceController
+      # By default JSONAPI::ResourceController provides most the standard
+      # behaviour, and in many cases this file may be left empty.
+    end
+
+    class QcFileProcessor < JSONAPI::Processor
+      # Validate that attributes contains both the filename and the contents of the QcFile.
+      def validate_required_attributes(attributes)
+        errors = []
+
+        filename = attributes[:filename]
+        errors += JSONAPI::Exceptions::ParameterMissing.new('filename').errors if filename.nil?
+
+        contents = attributes[:contents]
+        errors += JSONAPI::Exceptions::ParameterMissing.new('contents').errors if contents.nil?
+
+        [filename, contents, errors]
+      end
+
+      # Create a temporary file with the contents.
+      def create_tempfile(filename, contents)
+        # The filename for a Tempfile is passed as an array with the basename and extension.
+        # e.g. [ 'file', '.csv' ]
+        filename = [File.basename(filename, '.*'), File.extname(filename)]
```

Comment:
Maybe one of the two `filename` variables could be renamed. I was just a bit confused reading the comment on like 28, thinking it was refering to the filename parameter, instead of the variable created on line 30

Reply:
Good catch.  I agree.